lpcompiler.pas

```diff
@@ -2740,6 +2787,44 @@ function TLapeCompiler.ParseExpression(ReturnOn: EParserTokenSet = []; FirstNext
             Dec(InExpr);
           end;
 
+        tk_sym_BracketOpen:
+          if (_LastNode = _Var) then
+          begin
+            PopOpStack(op_Invoke);
+            if __IsProperty(VarStack.Top.resType()) then
+            begin
+              Expr := ResolveMethods(VarStack.Pop().FoldConstants(), True) as TLapeTree_ExprBase;
+              Prop := TLapeTree_InvokeProperty.Create(Expr, Self, getPDocPos());
+              if (Next() <> tk_sym_BracketClose) then
+              begin
+                Prop.addParam(EnsureExpression(ParseExpression([tk_sym_BracketClose, tk_sym_Comma], False)));
+                while True do
+                  case Tokenizer.Tok of
+                    tk_sym_BracketClose: Break;
+                    tk_sym_Comma:        Prop.addParam(EnsureExpression(ParseExpression([tk_sym_BracketClose, tk_sym_Comma])));
+                    else                 LapeException(lpeClosingBracketExpected, Tokenizer.DocPos);
+                  end;
+
+                if ParserTokenToOperator(Peek()) in AssignOperators then
```

Comment:
Yes, that might very well work. it just ended up like this as I wrote the array properties first, then moved on the the basic properties.